include/vfs.h

```diff
@@ -19,4 +20,13 @@ int do_dup2(thread_t *td, int old, int new);
 int do_mount(thread_t *td, const char *fs, const char *path);
 int do_getdirentries(thread_t *td, int fd, uio_t *uio, off_t *basep);
 
+typedef struct readdir_ops {
+  void *(*first)(vnode_t *v);
+  void *(*next)(void *entry);
+  unsigned (*namlen_of)(void *entry);
+  void (*convert)(void *entry, dirent_t *dir);
```

Comment:
My assumption was that `readdir_ops_t` functions should not be concerned with memory allocation. It's up to the algorithm to manage the memory. This will simplify things if we ever keep entries in some kind of cache, if that's the way to go...

I initially thought about replacing `namlen_of` with `reclen_of`, but then (in case of `initrd`) I'd need to run `strlen()` twice.